stdlib/src/memory/maybe_uninitialized.mojo

```diff
@@ -165,7 +181,8 @@ struct UnsafeMaybeUninitialized[ElementType: AnyType](CollectionElementNew):
         Args:
             other: The object to move.
         """
-        self.move_from(other.unsafe_ptr())
+        self = UnsafeMaybeUninitialized[MovableType](other._data^)
+        other = UnsafeMaybeUninitialized[MovableType]()
 
```

Comment:
I'm seeing that a bunch of the things that seemed wrong to me before, have been fixed.
This is now giving me:
`field 'other._data' destroyed out of the middle of a value, preventing the overall value from being destroyed`
Which is kind of what i expect, so thats good. I just have to think of a better way to do it now

Reply:
i tried it like this originally to avoid having `lit.ownership.mark_initialized()` everywhere, but maybe it makes more sense to do that anyways..